packages/svelte/src/compiler/phases/3-transform/client/visitors/IfBlock.js

```diff
@@ -11,21 +11,35 @@ export function IfBlock(node, context) {
 	context.state.template.push('<!>');
 
 	const consequent = /** @type {BlockStatement} */ (context.visit(node.consequent));
+	const consequent_id = context.state.scope.generate('consequent');
 
+	context.state.init.push(b.var(b.id(consequent_id), b.arrow([b.id('$$anchor')], consequent)));
+
+	let alternate_id;
+
+	if (node.alternate) {
+		const alternate = /** @type {BlockStatement} */ (context.visit(node.alternate));
+		alternate_id = context.state.scope.generate('alternate');
+		context.state.init.push(b.var(b.id(alternate_id), b.arrow([b.id('$$anchor')], alternate)));
+	}
+
+	/** @type {Expression[]} */
 	const args = [
 		context.state.node,
-		b.thunk(/** @type {Expression} */ (context.visit(node.test))),
-		b.arrow([b.id('$$anchor')], consequent)
+		b.arrow(
+			[b.id('$$branch')],
```

Comment:
maybe `$$render` instead of `$$branch`? that way it reads like an instruction — `$$render(consequent)`

```suggestion
			[b.id('$$render')],
```